### PR TITLE
Update EOL versions

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Comment by marcpopMSFT** [comment]
@richlander do we need to update this in servicing branches as well for 6, 7, and 8?

**Comment by richlander** [comment]
Yes, but I wanted to validate this in `main` first.

I see the pipeline timed out. Is that common?

File: .vsts-pr.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ stages:
     parameters:
       platform:
         name: 'Managed'
-        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8'
+        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0'
```
Comment (MichaelSimons) [inline .vsts-pr.yml:96]: Source build legs should continue to use centos as we don't ship a SB product via mariner.

```suggestion
        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
```

File: .vsts-ci.yml [hunk 2]
```diff
@@ -93,7 +93,7 @@ stages:
     parameters:
       platform:
         name: 'Managed'
-        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8'
+        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0'
```
Comment (MichaelSimons) [inline .vsts-ci.yml:96]: Source build legs should continue to use centos as we don't ship a SB product via mariner.

```suggestion
        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
```

**Comment by richlander** [comment]
Looks like this is now failing for the same reasons that other PRs are failing. What's the process in this repo for that problem? Wait until those issues are resolved?

**Comment by baronfel** [comment]
we merged a commit skipping the blocking test a few minutes ago, just need to rerun this PR.

**Comment by richlander** [comment]
Cool. Everything passes now.

**Comment by baronfel** [comment]
Just a heads up that the branches for servicing versions will fail for the same reason as this one did - we've staged the PRs to skip that failing test but cannot merge them until after branding happens, so your PRs will likely just be hanging out for a week or so until that occurs.

**Comment by richlander** [comment]
Got it. Will wait then.

**Comment by nagilson** [comment]
I think we can merge this now?

**Review by nagilson — APPROVED** [review]

**Review by baronfel — APPROVED** [review]

**Comment by richlander** [comment]
I will leave merging to you folks. Thanks!

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**